pkg/render/kube-controllers.go

```diff
@@ -423,6 +434,31 @@ func (c *kubeControllersComponent) controllersDeployment() *apps.Deployment {
 	return &d
 }
 
+// prometheusService creates a Service which exposes and endpoint on kube-controllers for
+// reporting Prometheus metrics.
+func (c *kubeControllersComponent) prometheusService() *v1.Service {
+	return &v1.Service{
+		TypeMeta: metav1.TypeMeta{Kind: "Service", APIVersion: "v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "calico-kube-controllers-metrics",
+			Namespace: common.CalicoNamespace,
+			Labels:    map[string]string{"k8s-app": "calico-kube-controllers"},
+		},
+		Spec: v1.ServiceSpec{
+			Selector: map[string]string{"k8s-app": "calico-kube-controllers"},
+			Type:     v1.ServiceTypeClusterIP,
+			Ports: []v1.ServicePort{
+				{
+					Name:       "metrics-port",
+					Port:       int32(c.metricsPort),
+					TargetPort: intstr.FromInt(int(9094)),
```

Comment:
Isn't the TargetPort the one that would need to change?
I'm not sure if we should set both to the values configured or just the TargetPort, probably just the TargetPort but whatever you think is best.